staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +285,54 @@ func (r *withRetry) After(ctx context.Context, request *Request, resp *http.Resp
 	}
 }
 
+func (r *withRetry) WrapPreviousError(currentErr error) error {
+	if currentErr == nil || r.previousErr == nil {
+		return currentErr
+	}
+	if !isContextError(currentErr) {
+		return currentErr
```

Comment:
if we have an `r.previousErr`, why don't we always return a `wrapPreviousError` even if it isn't a context error to show both?

Reply:
I have alwways wrapping the previous error if it is non nil and `curErr.Error() != prevErr.Error()`